geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-common/src/main/scala/org/locationtech/geomesa/fs/storage/common/metadata/FileBasedMetadata.scala

```diff
@@ -140,6 +140,25 @@ class FileBasedMetadata(
     }
   }
 
+  override def setPartitions(partitions: Seq[PartitionMetadata]): Unit = {
+    val map = new ConcurrentHashMap[String, PartitionFiles]
+    this.partitions.put(BoxedUnit.UNIT, map)
+    this.metadata.invalidateAll()
+
+    val configs = partitions.map { partition =>
+      val action = PartitionAction.Add
+      val envelope = partition.bounds.map(b => EnvelopeConfig(b.envelope)).getOrElse(Seq.empty)
+      val config = PartitionConfig(partition.name, action, partition.files, partition.count, envelope, System.currentTimeMillis())
+      // note: side effects in map
```

Comment:
Can you say more about the side effects here?  I'm not seeing which of these calls has a side effect.

Reply:
oh, it's the 2 lines after this comment